src/scene/materials/standard-material.js

```diff
@@ -490,7 +486,7 @@ class StandardMaterial extends Material {
         if (!this.emissiveMap || this.emissiveTint) {
             this._setParameter('material_emissive', getUniform('emissive'));
         }
-        if (this.emissiveMap) {
+        if (this.emissiveIntensity !== 1) {
```

Comment:
does this mean that when the user changes emissiveIntensity from 1 to other value, the shader needs to be recompiled?

Reply:
yep :( there are a bunch of such micro-optimisations in the shaders.